### PR TITLE
Fix embedded PDB support on big-endian systems

### PR DESCRIPTION
**Pull request by uweigand**
* Use little-endian byte order when writing integers to the buffer
  in pdbGetCvDebugInfo and pdbGetEmbeddedPdbDebugInfo

This showed up as a bug when bootstrapping the fsharp compiler natively on a big-endian machine (linux-s390x - IBM Z mainframe - which we're working on enabling for .NET 6).  The compiler would create a .dll file with an invalid embedded PDB header, which would then cause the runtime to abort with a failed assertion when running that .dll.   With this patch, native bootstrap of fsharp is successful on linux-s390x.

The implementation of "i32AsBytes" is copied from fsharp/absil/il.fs.

### CODE REVIEW

File: src/fsharp/absil/ilwritepdb.fs [hunk 1]
```diff
@@ -171,18 +171,25 @@ let checkSum (url: string) (checksumAlgorithm: HashAlgorithm) =
 //---------------------------------------------------------------------
 // Portable PDB Writer
 //---------------------------------------------------------------------
+
+let b0 n = (n &&& 0xFF)
```
[the remark this answers is not in the document]
Comment (dsyme) [reply src/fsharp/absil/ilwritepdb.fs:175]: `i32AsBytes` could be moved from il.fs into lib.fs too, there are a few duplications of these

**Review by dsyme — APPROVED** [review]

File: src/fsharp/absil/ilwritepdb.fs (hunk 1 above)
[the remark this answers is not in the document]
Comment (uweigand) [reply src/fsharp/absil/ilwritepdb.fs:175]: I see.   Just to clarify, that module seems to contain the b0 .. b3 functions, but not i32AsBytes, so the latter would have to remain, right?

**Review by vzarytovskii — APPROVED** [review]

File: src/fsharp/absil/ilwritepdb.fs (hunk 1 above)
[the remark this answers is not in the document]
Comment (dsyme) [reply src/fsharp/absil/ilwritepdb.fs:175]: It's actually OK, this is not the only duplication, we can clean that up later I think

[the remark this answers is not in the document]
Comment (dsyme) [reply src/fsharp/absil/ilwritepdb.fs:175]: I merged - if you'd like to submit a PR consolidating `b0` etc that would be cool

**Comment by uweigand** [comment]
Thanks!   I'll prepare a separate PR to clean up the duplication.

As to this PR - it would be great to have this in .NET 6 as the problem prevents bootstrapping the compiler on s390x (and therefore prevents a source-build of the whole SDK to go through).   Does this mean the PR will have to be backported to the dev17 branch, and if so, would that be possible?

**Comment by uweigand** [comment]
> As to this PR - it would be great to have this in .NET 6 as the problem prevents bootstrapping the compiler on s390x (and therefore prevents a source-build of the whole SDK to go through). Does this mean the PR will have to be backported to the dev17 branch, and if so, would that be possible?

Never mind, I see this has already happened automatically.

> Thanks! I'll prepare a separate PR to clean up the duplication.
 
Actually, I could use a bit of advice here.  It seems the duplication is not quite identical: the "b0" .. "b3" functions in lib.fs return "int" while a number of the duplicates return "byte".  (Note that all copies of the 64-bit "dw0" .. "dw7" routines return "byte".)

If those are to be merged, should the result preferably be "int" or "byte" ?   Either way seems to require a number of follow-on changes to fix the fall-out.

**Pull request merged**